BondageClub/Screens/Online/ChatRoom/ChatRoom.js

```diff
@@ -1539,142 +1545,47 @@ function ChatRoomKeyDown(event) {
-		} else {
-			var WhisperTarget = null;
-			for (let C = 0; C < ChatRoomCharacter.length; C++)
-				if (ChatRoomTargetMemberNumber == ChatRoomCharacter[C].MemberNumber)
-					WhisperTarget = ChatRoomCharacter[C];
-			if (msg != "" && !((ChatRoomTargetMemberNumber != null || m.indexOf("(") >= 0) && Player.ImmersionSettings && (Player.ImmersionSettings.BlockGaggedOOC && (!(Player.Effect.includes("HideRestraints") && Player.Effect.includes("VRAvatars")) || !WhisperTarget || !WhisperTarget.Effect.includes("VRAvatars"))) && !Player.CanTalk())) {
-				if (ChatRoomTargetMemberNumber == null) {
-					// Regular chat
-					ServerSend("ChatRoomChat", { Content: msg, Type: "Chat" });
-					ChatRoomStimulationMessage("Gag");
-				} else {
-					// The whispers get sent to the server and shown on the client directly
-					ServerSend("ChatRoomChat", { Content: msg, Type: "Whisper", Target: ChatRoomTargetMemberNumber });
-					var TargetName = "";
-					if (WhisperTarget) TargetName = WhisperTarget.Name;
-
-					var div = document.createElement("div");
-					div.setAttribute('class', 'ChatMessage ChatMessageWhisper');
-					div.setAttribute('data-time', ChatRoomCurrentTime());
-					div.setAttribute('data-sender', Player.MemberNumber.toString());
-					div.innerHTML = TextGet("WhisperTo") + " " + TargetName + ": " + msg;
-
-					var Refocus = document.activeElement.id == "InputChat";
-					var ShouldScrollDown = ElementIsScrolledToEnd("TextAreaChatLog");
-					if (document.getElementById("TextAreaChatLog") != null) {
-						document.getElementById("TextAreaChatLog").appendChild(div);
-						if (ShouldScrollDown) ElementScrollToEnd("TextAreaChatLog");
-						if (Refocus) ElementFocus("InputChat");
-					}
-				}
-			} else {
-				// Throw an error message
-				ChatRoomMessage({ Content: "ChatRoomBlockGaggedOOC", Type: "Action", Sender: Player.MemberNumber });
-			}
-		}
-		// Clears the chat text message
-		ElementValue("InputChat", "");
+/**
+ * Sends message to user with HTML tags
+ * @param {string} Content - InnerHTML for the message
+ * @param {number} [Timeout] - total time to display the message in ms
+ * @returns {void} - Nothing
+ */
+function ChatRoomSendLocal(Content, Timeout) {
+	ChatRoomMessage({
+		Sender: Player.MemberNumber,
+		Type: "LocalMessage",
+		Content, Timeout,
+	});
+}
 
+/**
+ * Removes (*) (/me) (/action) then sends message as emote
+ * @param {string} msg - Emote message
+ * @returns {void} - Nothing
+ */
+function ChatRoomSendEmote(msg) {
+	if (Player.ChatSettings.MuStylePoses) msg = msg.replace(":", "");
+	else if (msg.indexOf("**") == 0) {
+		msg = "*" + msg.replace(/\*/g, "");
+	} else {
+		msg = msg.replace(/\*/g, "");
+		if (msg.startsWith(CommandsKey + "me")) msg = msg.replace(CommandsKey + "me", "");
+		if (msg.startsWith(CommandsKey + "action")) msg = msg.replace(CommandsKey + "action", "*");
```

Comment:
```suggestion
		if (msg.startsWith(CommandsKey + "me ")) msg = msg.replace(CommandsKey + "me ", "");
		if (msg.startsWith(CommandsKey + "action ")) msg = msg.replace(CommandsKey + "action ", "*");
```
This is translating "/action word" into "* word*" instead of "*word*".